From 7613472953ad8b3061ca80fdd94658cd234d8c6f Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Thu, 15 Dec 2011 15:39:31 +0800
Subject: [PATCH 0749/2463] ENGR00170350 dvi detect: request cable detect pin
 when real init

request cable detect pin when real init

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_arm2.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_arm2.c b/arch/arm/mach-mx6/board-mx6q_arm2.c
index 54c7fc0..b0f45f0 100644
--- a/arch/arm/mach-mx6/board-mx6q_arm2.c
+++ b/arch/arm/mach-mx6/board-mx6q_arm2.c
@@ -648,6 +648,10 @@ static void ddc_dvi_init(void)
 {
 	/* enable DVI I2C */
 	gpio_set_value(MX6Q_ARM2_DISP0_I2C_EN, 1);
+
+	/* DISP0 Detect */
+	gpio_request(MX6Q_ARM2_DISP0_DET_INT, "disp0-detect");
+	gpio_direction_input(MX6Q_ARM2_DISP0_DET_INT);
 }
 
 static int ddc_dvi_update(void)
@@ -1500,10 +1504,6 @@ static void __init mx6_board_init(void)
 	imx_asrc_data.asrc_audio_clk = clk_get(NULL, "asrc_serial_clk");
 	imx6q_add_asrc(&imx_asrc_data);
 
-	/* DISP0 Detect */
-	gpio_request(MX6Q_ARM2_DISP0_DET_INT, "disp0-detect");
-	gpio_direction_input(MX6Q_ARM2_DISP0_DET_INT);
-
 	/* DISP0 Reset - Assert for i2c disabled mode */
 	gpio_request(MX6Q_ARM2_DISP0_RESET, "disp0-reset");
 	gpio_direction_output(MX6Q_ARM2_DISP0_RESET, 0);
-- 
1.7.7.4

